.github/workflows/ci.yml

```
@@ -13,6 +13,30 @@ jobs:
 
   # Note that vcpkg dependencies takes the majority of the build time.
   # We cache them using GitHub Actions cache and export, making the scripts below a bit more complex.
+  check-format:
+    if: github.event_name == 'push' || github.event.pull_request.head.repo.id != github.event.pull_request.base.repo.id
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup .NET Core SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: ${{ env.dotnet_5_version }} 
+      - name: Code formating check
+        run: |
+          dotnet tool restore
+          dotnet jb cleanupcode "csharp" "csharp.test" "csharp.benchmark" --profile="Built-in: Reformat Code" --settings="ParquetSharp.DotSettings" --dotnetcoresdk=${{ env.dotnet_5_version }} --verbosity=WARN
+          check=$(git diff --name-only | wc -l)
+          if [ $check -eq 0 ]
+          then
+            echo "Check passed"
+            exit 0
+          else
+            echo "Code format check - FAILED"
+            git diff --name-only | echo "::error file=$(awk '{print $1}') ::Code format check failed"
+            exit 1
+          fi
```

This will not output the proper error. Instead there should be one error per badly-formatted file.
We can also drop the positive case and call `git diff` just once with something like this:

```suggestion
          files=($(git diff --name-only))
          if [ ${#files[@]} -gt 0 ]
          then
            for file in $files; do echo "::error file=$file::Code format check failed"; done
            exit 1
          fi
```

---

pure beauty 🙏 👍